### PR TITLE
os/board/rtl8730e: Integrate BT FW patch to resource bin

### PR DESCRIPTION
**Pull request by edwakuwaku**
- Currently, the BT FW patch is independently flashed to a specific region in external address. This could cause a failure in SW update, because SW update support dual region
- This commit changes the flow to use ioctl (ie. access to resource bin in ext flash) to bringup BT, so that any future update on BT FW patch will be included in SW update as well

### CODE REVIEW

File: build/configs/rtl8730e/rtl8730e_make_bin.sh [hunk 1]
```diff
@@ -19,6 +19,7 @@
 THIS_PATH=`test -d ${0%/*} && cd ${0%/*}; pwd`
 TOP_PATH=${THIS_PATH}/../../..
 CONFIG=${TOP_PATH}/os/.config
+RESOURCE=${TOP_PATH}/resource
```
Comment (edwakuwaku) [inline build/configs/rtl8730e/rtl8730e_make_bin.sh:22]: Hello @jeongarmy , as discussed, after you complete the separation for chipset specific files in resource directory, I will change this file path to` /resource/rtl8730e/...bin`
Thank you~

**Comment by sunghan-chang** [comment]
@edwakuwaku Let us verify this again and then merge this. It will be next week.

**Comment by jeongarmy** [comment]
I verified it runs well. Could you remove this line about bt binary?
https://github.com/Samsung/TizenRT/blob/master/build/configs/rtl8730e/board_metadata.txt#L26

**Comment by jeongarmy** [comment]
Could you move making and copying rtk binary to Makefile instead of these files?
As you know, Internal repositories for commercialization are separated and the final binaries are made in product level.
Currently these scripts are run in product build, not kernel build and this files are put in product/resource folder.
So I wanted rtw binary to be created and put in resource folder of kernel repo because this file is kernel's resource.
Could you please check this?

**Comment by edwakuwaku** [comment]
> Could you move making and copying rtk binary to Makefile instead of these files? As you know, Internal repositories for commercialization are separated and the final binaries are made in product level. Currently these scripts are run in product build, not kernel build so these files are put in product/resource folder. So I wanted rtw binary to be created and put in resource folder of kernel repo when kernel build because this file is kernel's resource. Could you please check this?

I will check and apply necessary changes. Besides that, could you let me know the final decision of path dir for binary placement? Will it be resource/xxx.bin or resource/rtl8730e/xxx.bin?

**Comment by jeongarmy** [comment]
> > Could you move making and copying rtk binary to Makefile instead of these files? As you know, Internal repositories for commercialization are separated and the final binaries are made in product level. Currently these scripts are run in product build, not kernel build so these files are put in product/resource folder. So I wanted rtw binary to be created and put in resource folder of kernel repo when kernel build because this file is kernel's resource. Could you please check this?
> 
> I will check and apply necessary changes. Besides that, could you let me know the final decision of path dir for binary placement? Will it be resource/xxx.bin or resource/rtl8730e/xxx.bin?

I think the second one looks better. thank you!

**Comment by sunghan-chang** [comment]
@jeongarmy Let's verify this ASAP and merge. Could you let me know test result?

**Comment by sunghan-chang** [comment]
@edwakuwaku FYI. This could be merged after Thu.

File: os/board/rtl8730e/src/Makefile [hunk 2]
```diff
@@ -213,16 +218,31 @@ $(COBJS) $(LINKOBJS): %$(OBJEXT): %.c
 
 libboard$(LIBEXT): $(OBJS)
 	$(call ARCHIVE, $@, $(OBJS))
+ifeq ($(CONFIG_SECOND_FLASH_PARTITION),y)
```
Comment (ewoodev) [inline os/board/rtl8730e/src/Makefile:221]: Please use CONFIG_RESOURCE_FS

**Comment by edwakuwaku** [comment]
Hi @sunghan-chang @ewoodev , now CONFIG_RTK_DATA_BINARY_TO_EXT_FLASH depends on CONFIG_SECOND_FLASH_PARTITION and CONFIG_RESOURCE_FS.
CONFIG_RTK_DATA_BINARY_TO_EXT_FLASH  is disabled by default.

**Comment by sunghan-chang** [comment]
@edwakuwaku There are two conflicts. Could you resolve them?

File: os/board/rtl8730e/src/Makefile [hunk 3]
```diff
@@ -213,16 +218,31 @@ $(COBJS) $(LINKOBJS): %$(OBJEXT): %.c
...
+	@if [ -d $(RESOURCE_TARGET_FOLDER) ]; then echo "$(RESOURCE_TARGET_FOLDER) exist"; else $(MKDIR) -p $(RESOURCE_TARGET_FOLDER); fi
+	$(Q) mv -f $(BUILD_DIR)/rtk_ext_flash_data.bin $(RESOURCE_TARGET_FOLDER)/
+	@echo "========== Current config support external flash, binary created and moved to path /resource/rtl8730e/rtk_ext_flash_data.bin! =========="
+endef
+
 clean:
 	$(call DELFILE, libboard$(LIBEXT))
 	$(call CLEAN)
+ifeq ($(CONFIG_RTK_DATA_BINARY_TO_EXT_FLASH),y)
+	@rm -rf $(RESOURCE_TARGET_FOLDER)
```
Comment (sunghan-chang) [inline os/board/rtl8730e/src/Makefile:244]: Let's use the same function for the same purpose.
```
$(call DELDIR, $(RESOURCE_TARGET_FOLDER))
```
If it's the folder, then DELDIR. 
DELFILE for file

**Comment by sunghan-chang** [comment]
@jeongarmy please check this by tomorrow.

File: os/board/rtl8730e/src/component/bluetooth/driver/platform/amebad2/hci/hci_platform.c [hunk 4]
```diff
@@ -939,58 +932,46 @@ static uint32_t hci_platform_parse_patch(uint8_t *p_buf, HCI_PATCH_NODE *p_patch
...
+	}
 
-    return rtlbt_fw_ptr;
+errout:
+	/* If any step failed, BT cannot be initialized */
+	if (ret) {
+		dbg("BT init failed, please check line %d!\n", ret);
+		ASSERT(0);
+	}
+	return rtlbt_fw_ptr;
```
[the remark this answers is not in the document]
Comment (sunghan-chang) [reply os/board/rtl8730e/src/component/bluetooth/driver/platform/amebad2/hci/hci_platform.c:974]: @ziyik Could you check this comment? I don't know @edwakuwaku can make change for this. If not, your team should get this PR and should make a change.
```
git remote add upstream https://github.com/Samsung/TizenRT.git
git fetch upstream pull/6303/head:pr-6303
```
make a change and then push

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**